static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -28,66 +29,48 @@ function findEnd(data) {
   return data.length - 1;
 }
 
-function getHighest(data) {
-  let highestIndex = 0;
-  let highestValue = 0;
-  for (let i = 0, ii = data.length ; i < ii; i += 1) {
-    if (data[i] > highestValue) {
-      highestValue = data[i];
-      highestIndex = i;
-    }
-  }
-
-  return {
-    index: highestIndex,
-    value: highestValue
-  };
-}
+const labelPosition = function(graphEl, labelText, points) {
+  const start = findStart(points);
+  const end = findEnd(points);
 
-const labelPosition = function(graphEl, labelText, data, isFirst, isLast) {
-  const start = findStart(data);
-  const end = findEnd(data);
-  const highestIndex = getHighest(data).index;
-
-  const labelClass = labelText.replace('.', '-');
+  const labelClass = labelText.replace(/\W+/g, '-');
   const area = graphEl.querySelector(`.bb-areas-${labelClass}`);
   const bbox = area.getBBox();
-  const sliceWidth = Math.round(bbox.width / data.length);
 
-  const width = sliceWidth * (end - start);
-  let leftPadding = (width / 12);
-  
-  if (isLast && highestIndex > end - 3) {
-    leftPadding = -(width / 12);
-  }
+  const pointsWidth = (bbox.width / points.length) * (end - start);
 
-  const highestLeft = parseInt(
-    graphEl.querySelector(`.bb-event-rect-${highestIndex}`).getAttribute('x'),
-    10
-  );
+  if (bbox.height < 20) {
+    return;
+  }
 
   let label = document.querySelector(`[data-version="${labelText}"]`);
   if (!label) {
     label = document.createElementNS('http://www.w3.org/2000/svg', 'text');
+    label.setAttribute('class', 'snapcraft-metrics__active-devices-label');
     label.setAttribute('data-version', labelText);
     label.setAttribute('stroke', '#000');
-    label.setAttribute('x', highestLeft + leftPadding);
-    label.setAttribute('y', bbox.y + (bbox.height / 2));
     const labelSpan = document.createElementNS('http://www.w3.org/2000/svg', 'tspan');
     labelSpan.innerHTML = labelText;
+    labelSpan.setAttribute('text-anchor', 'middle');
     label.appendChild(labelSpan);
     area.appendChild(label);
-  } else {
-    label.setAttribute('x', highestLeft + leftPadding);
-    label.setAttribute('y', bbox.y + (bbox.height / 2));
   }
+  label.setAttribute('x', pointsWidth / 2);
+  label.setAttribute('y', bbox.y + (bbox.height / 2));
 };
 
-const positionLabels = function(el, activeDevices) {
+const positionLabels = function(el, activeDevices, activeDevicesMetrics) {
   activeDevices.forEach((points, index) => {
     let _points = points.slice(0);
     const version = _points.shift();
-    labelPosition(el, version, _points, index === 0, index === activeDevices.length - 1);
+    labelPosition(
```

Comment:
It seems that `labelPosition` was changed and now only takes 3 parameters, not 6.

Reply:
It's good to have a buffer just incase.... I'll remove them.